service/mantle/shopify/ApiServices.xml

```diff
@@ -458,17 +458,36 @@
         </in-parameters>
         <out-parameters>
             <parameter name="collectData"/>
+            <parameter name="error"/>
         </out-parameters>
         <actions>
             <entity-find-one entity-name="mantle.product.category.ProductCategory" value-field="productCategory">
                 <field-map field-name="productCategoryId" from="productCategoryId"/>
             </entity-find-one>
+            
+            <if condition="!productCategory">
+                <set field="error" value="true"/>
+                <message type="warning">The product category ${productCategoryId} does not exist</message>
+                <return/>
+            </if>
+            <if condition="productCategory.pseudoId == productCategory.productCategoryId">
+                <set field="error" value="true"/>
+                <message type="warning">The product category ${productCategoryId} does not have shopifyId</message>

```

Comment:
This will generate conflicts with the PR 17, define which one you want to have in the code.